src/client/src/Client.c

```diff
@@ -55,7 +57,14 @@ STATUS checkIntermittentProducerCallback(UINT32 timerId, UINT64 currentTime, UIN
         for (i = 0; i < pKinesisVideoClient->deviceInfo.streamCount; i++) {
             if (NULL != pKinesisVideoClient->streams[i]) {
                 pCurrStream = pKinesisVideoClient->streams[i];
-                // Lock the Stream
+                pFrameOrderCoordinator = pCurrStream->pFrameOrderCoordinator;
+                if (pCurrStream->streamInfo.streamCaps.frameOrderingMode != FRAME_ORDER_MODE_PASS_THROUGH) {

```

Comment:
Curious: What happens in the where it is `FRAME_ORDER_MODE_PASS_THROUGH`? Why don't we require locking for that case?

Reply:
If you follow the code flow in the putKinesisVideoFrame API right before we acquire the frame order coordinator lock we check for the mode.  If mode is the pass through we short circuit and call putFrame immediately.  The reason for this is this mode indicates we have only 1 track.  And the whole point of frame order coordinate is to coordinate between multiple tracks.

Reply:
Ah! Ok. Makes sense now. My bad.